src/templates/components/image/index.js

```diff
@@ -1,33 +1,61 @@
 import { h } from 'preact';
 
-const ImageSource=({fileName, data}) => {
-	return <source srcset={`${fileName.split('.')[0]}${data.sizeID}.${data.format}${data.query}`} media={`(min-width:${data.minWidth}px)`} />
-}
-
+/**
+ * Responsive image component, using the <picture> element
+ * @param {string} alt - Image alt tag
+ * @param {string} decoding='async' - 'async', 'sync', 'auto'
+ * @param {string} loading='lazy' - 'lazy', 'eager'
+ * @param {string} src - Reference to local/relative or remote image URI
+ * @param {string} className - Class name
+ * @param {array} [sources]	- Array of Objects for <picture> —> <source> attributes {src, media}
+ **/
 const Image=({
-	componentClass,
-	imageClass,
+	className,
 	alt,
-	fileName,
-	altSizes,
+	src,
+	sources,
 	decoding='async',
 	loading='lazy'
-	}) => 
+	}) =>
 
-<picture class={`image ${componentClass ? componentClass : ''}`}>
+<picture class={`${className ? className : ''}`}>
 
 	{
-		altSizes && altSizes.map(size => ImageSource({fileName:fileName, data:size}))
+		sources && sources.map(srcData => <source srcset={srcData.src} media={srcData.media} />)
 	}
 
 	<img 
 		alt={alt}
-		class={`${imageClass ? imageClass : ''}`}
+		class={`${className ? className : ''}`}
```

Comment:
Hi Nick!  I think for this bit, now that we're not adding a default class, this could just be class={className} in the style of your other attributes.  That way we don't get an empty class being written out if it's not used

Reply:
Of course. Thank you.